rescan.go

```diff
@@ -120,6 +123,16 @@ func NotificationHandlers(ntfn rpcclient.NotificationHandlers) RescanOption {
 	}
 }
 
+// UtxoScannerProgressHandler specified a handler to be used when the utxo
+// scanner reports its progress.
```

Comment:
should add a note that this should be non-blocking, otherwise it might stall progress.